web/viewer.js

```diff
@@ -211,8 +211,20 @@ function webViewerLoad() {
       // set various `AppOptions`, by dispatching an event once all viewer
       // files are loaded but *before* the viewer initialization has run.
       const event = document.createEvent("CustomEvent");
-      event.initCustomEvent("webviewerloaded", true, true, {});
-      document.dispatchEvent(event);
+      event.initCustomEvent("webviewerloaded", true, true, {
+        source: window,
+      });
+      try {
+        // Attempt to dispatch the event at the embedding `document`,
+        // in order to support cases where the viewer is embedded in
+        // a *dynamically* created <iframe> element.
+        parent.document.dispatchEvent(event);
+      } catch (ex) {
+        // The viewer could be in e.g. a cross-origin <iframe> element,
+        // fallback to dispatching the event at the current `document`.
+        console.error(`webviewerloaded: ${ex}`);
```

Comment:
Since there is a valid case in which this can happen and we have a fallback to handle it, do we need the error log here?

Reply:
In this special case you won't be able to (easily) listen for the "webviewerloaded" event if using a *dynamically* created <iframe> element, and there's basically no reasonable way to fix the referenced issue in that case unfortunately.

By at least logging the error here, we should hopefully save time debugging if someone submits a future issue specifically for a cross-origin and dynamically created <iframe> element for the viewer.